model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java

```diff
@@ -189,7 +189,10 @@ public AuthenticatedClientSessionModel createClientSession(RealmModel realm, Cli
     public UserSessionModel createUserSession(String id, RealmModel realm, UserModel user, String loginUsername, String ipAddress,
                                               String authMethod, boolean rememberMe, String brokerSessionId, String brokerUserId, UserSessionModel.SessionPersistenceState persistenceState) {
         if (id == null) {
-            id = keyGenerator.generateKeyString(session, sessionTx.getCache());
+            do {
+                id = keyGenerator.generateKeyString(session, sessionTx.getCache());
+                // prevent potential duplicate ID generation
+            } while (sessionTx.get(id) != null);
```

Comment:
uh!? What's the probability of a conflict? 1 in 2^122? It seems unnecessary.

Reply:
OK, I'll leave that for a separate PR